experimental/http_logger/http_logger_test.go

```diff
@@ -0,0 +1,77 @@
+package httplogger_test
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/e2e/storage"
+	httplogger "github.com/grafana/grafana-plugin-sdk-go/experimental/http_logger"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHTTPLogger(t *testing.T) {
+	t.Run("saved file should match example", func(t *testing.T) {
+		c, f := setup(true)
+		res, err := c.Get("http://example.com")
+		require.NoError(t, err)
+		defer res.Body.Close()
+		require.Equal(t, http.StatusOK, res.StatusCode)
+		b, err := ioutil.ReadAll(res.Body)
+		require.NoError(t, err)
+		require.Equal(t, "OK", string(b))
+		expected := storage.NewHARStorage("testdata/example.har")
+		actual := storage.NewHARStorage(f.Name())
+		require.Equal(t, 1, len(actual.Entries()))
+		require.Equal(t, expected.Entries()[0].Request, actual.Entries()[0].Request)
+		require.Equal(t, expected.Entries()[0].Response, actual.Entries()[0].Response)
+		har, err := ioutil.ReadFile(f.Name())
+		require.NoError(t, err)
+		require.Greater(t, len(har), 0)
+	})
+
+	t.Run("file should not be created by storage if http logging is disabled", func(t *testing.T) {
+		c, f := setup(false)
+		res, err := c.Get("http://example.com")
+		require.NoError(t, err)
+		defer res.Body.Close()
+		require.Equal(t, http.StatusOK, res.StatusCode)
+		b, err := ioutil.ReadAll(res.Body)
+		require.NoError(t, err)
+		require.Equal(t, "OK", string(b))
+		actual := storage.NewHARStorage(f.Name())
+		require.Equal(t, 0, len(actual.Entries()))
+		_, err = os.Stat(f.Name())
+		require.Equal(t, true, errors.Is(err, os.ErrNotExist))
+	})
+}
+
+func setup(enabled bool) (*http.Client, *os.File) {
+	f, err := os.CreateTemp("", "example_*.har")
+	defer os.Remove(f.Name())
+	if err != nil {
+		panic(err)
+	}
+	h := httplogger.
+		NewHTTPLogger("example-plugin-id", &fakeRoundTripper{}).
+		WithPath(f.Name()).
```

Comment:
With this builder pattern, won't `NewHTTPLogger` call create a new temp `.har` file, wrap it in a store and get initialized, only to be immediately thrown away with `WithPath()`?

Reply:
Nice one 👍  If it's for mostly tests, could we un-export it so it's not part of the API (will have to change the test to be in the same  `httplogger` package though)? Or alternatively have something like this:

```golang
// NewHTTPLogger creates a new HTTPLogger.
func NewHTTPLogger(pluginID string, proxied http.RoundTripper) *HTTPLogger {
	return newHttpLogger(pluginID, defaultPath(pluginID), proxied)
}

// used for tests
func newHttpLogger(pluginID, path string, proxied http.RoundTripper) *HTTPLogger {
	s := storage.NewHARStorage(path)
	f := fixture.NewFixture(s)

	return &HTTPLogger{
		pluginID: pluginID,
		proxied:  proxied,
		fixture:  f,
		enabled:  defaultEnabledCheck,
	}
}
```

Reply:
i was able to remove this in [4fd753](https://github.com/grafana/grafana-plugin-sdk-go/pull/471/commits/e4fd753666a40c8afb47059b2cc721eaca9d074d). it's testing using the env vars now, so it's actually closer to real world usage